editor/debugger/script_editor_debugger.cpp

```diff
@@ -350,536 +350,595 @@ void ScriptEditorDebugger::_select_thread(int p_index) {
+void ScriptEditorDebugger::_msg_filesystem_update_file(uint64_t p_thread_id, const Array &p_data) {
+	ERR_FAIL_COND(p_data.is_empty());
+	if (EditorFileSystem::get_singleton()) {
+		EditorFileSystem::get_singleton()->update_file(p_data[0]);
+	}
+}
 
-		if (p_msg == "servers:profile_frame") {
-			profiler->add_frame_metric(metric, false);
-		} else {
-			profiler->add_frame_metric(metric, true);
-		}
-	} else if (p_msg == "request_quit") {
-		emit_signal(SNAME("stop_requested"));
-		_stop_and_notify();
-	} else if (p_msg == "remote_nodes_clicked") {
-		ERR_FAIL_COND(p_data.is_empty());
-		EditorDebuggerRemoteObjects *objs = inspector->set_objects(p_data);
-		if (objs) {
-			EditorDebuggerNode::get_singleton()->stop_waiting_inspection();
+void ScriptEditorDebugger::_msg_evaluation_return(uint64_t p_thread_id, const Array &p_data) {
+	expression_evaluator->add_value(p_data);
+}
 
-			emit_signal(SNAME("remote_objects_updated"), objs);
-			emit_signal(SNAME("remote_tree_select_requested"), objs->remote_object_ids.duplicate());
-		}
-	} else if (p_msg == "remote_nothing_clicked") {
-		EditorDebuggerNode::get_singleton()->stop_waiting_inspection();
+void ScriptEditorDebugger::_msg_window_title(uint64_t p_thread_id, const Array &p_data) {
+	ERR_FAIL_COND(p_data.size() != 1);
+	emit_signal(SNAME("remote_window_title_changed"), p_data[0]);
+}
 
-		emit_signal(SNAME("remote_tree_clear_selection_requested"));
-	} else if (p_msg == "remote_selection_invalidated") {
-		ERR_FAIL_COND(p_data.is_empty());
-		inspector->invalidate_selection_from_cache(p_data[0]);
-	} else if (p_msg == "show_selection_limit_warning") {
-		EditorToaster::get_singleton()->popup_str(vformat(TTR("Some remote nodes were not selected, as the configured maximum selection is %d. This can be changed at \"debugger/max_node_selection\" in the Editor Settings."), EDITOR_GET("debugger/max_node_selection")), EditorToaster::SEVERITY_WARNING);
-	} else if (p_msg == "performance:profile_names") {
-		Vector<StringName> monitors;
-		monitors.resize(p_data.size());
-		for (int i = 0; i < p_data.size(); i++) {
-			ERR_FAIL_COND(p_data[i].get_type() != Variant::STRING_NAME);
-			monitors.set(i, p_data[i]);
-		}
-		performance_profiler->update_monitors(monitors);
-	} else if (p_msg == "filesystem:update_file") {
-		ERR_FAIL_COND(p_data.is_empty());
-		if (EditorFileSystem::get_singleton()) {
-			EditorFileSystem::get_singleton()->update_file(p_data[0]);
-		}
-	} else if (p_msg == "evaluation_return") {
-		expression_evaluator->add_value(p_data);
-	} else if (p_msg == "window:title") {
-		ERR_FAIL_COND(p_data.size() != 1);
-		emit_signal(SNAME("remote_window_title_changed"), p_data[0]);
+void ScriptEditorDebugger::_parse_message(const String &p_msg, uint64_t p_thread_id, const Array &p_data) {
+	emit_signal(SNAME("debug_data"), p_msg, p_data);
+
+	ParseMessageFunc *fn_ptr = parse_message_handlers->getptr(p_msg);
+	if (fn_ptr) {
+		(this->**fn_ptr)(p_thread_id, p_data);
```

Comment:
Just for my curiosity: We don't usually do explicit `this->` dereference, is this necessary here or just a stylistic choice?

Reply:
@akien-mga I found a few examples, such as this:

https://github.com/godotengine/godot/blob/d6da30e9c77825530702bd37a557a42f8c86d814/modules/gdscript/gdscript_parser.cpp#L2591